ee/tuf/autoupdate.go

```diff
@@ -298,6 +289,16 @@ func (ta *TufAutoupdater) Do(data io.Reader) error {
 		return nil
 	}
 
+	if time.Now().Before(ta.initialDelayEnd) && !updateRequest.BypassInitialDelay {
+		ta.slogger.Log(context.TODO(), slog.LevelWarn,
+			"received update request during initial delay, discarding",
+			"initial_delay_end", ta.initialDelayEnd.UTC().Format(time.RFC3339),
+		)
+		// We don't return an error because there's no need for the actionqueue to retry this request --
+		// we're going to perform an autoupdate check as soon as we exit the delay anyway.
+		return nil
+	}
+
```

Comment:
Oh, will we see an issue with the osquery restart func being called before we've actually started an osquery instance?

Reply:
this is triggered inline (blocking) before any of the rungroups are started, that was the only way I could find to avoid the osq runner erroring out and restarting everything. so I think we're safe there, if that's what you mean?

Reply:
justification was there's really no reason to avoid hanging up the main thread or bypassing initial delay when nothing will work anyway. and `launchOsqueryInstance` does a fresh path lookup at runtime which in testing handled this all perfectly already

Reply:
Yeah, was asking if it was safe to call the restart func before the rungroup started! Took a quick look -- do we need a `r.instance != nil` check in the restart func? https://github.com/kolide/launcher/blob/main/pkg/osquery/runtime/runner.go#L207

Reply:
Thank you!

Reply:
🤦 sorry I see what you meant, yeah it seems to be okay! Thank you!